include/power/pm_policy.h

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2018 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _PM_POLICY_H_
+#define _PM_POLICY_H_
+
+#include <power/power.h>
+
+#include <string.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Define the base struct necessary for a pm_policy.
+ *
+ * Different policies can have their own struct type according with
+ * their necessity, the only requirements is the struct contains this
+ * macro in the beginning of the declaration and be statically defined
+ * using @c PM_POLICY_DEFINE.
+ */
+#define PM_POLICY_BASE \
+	struct { \
+		const char *name; \
+		const size_t len; \
+		enum power_states (*next_state_get)(struct pm_policy *policy,\
+						    int32_t ticks); \
+		bool (*low_power_devices_get)(struct pm_policy *policy, \
+					      enum power_states state); \
+	}
+
+/**
+ * @brief PM Policy definition
+ *
+ * Store information about a power management policy.
+ */
+struct pm_policy {
+	PM_POLICY_BASE;
```

Comment:
While we're talking: I'd really rather not see the common parts of the structure provided by a define.  It should be a standalone structure, which will be the first member of  a specific policy structure, or accessed by `CONTAINER_OF`.  Field declarations in a macro are really hard to read.

Reply:
I just changed to something simpler ... Just use a basic type wouldn't work as I wanted, since it was supposed to be statically allocated by the `PM_POLICY_DEFINE` macro. I simplified it and now it is not possible to extends the type.